scipy/cluster/tests/test_vq.py

```diff
@@ -330,9 +333,9 @@ def test_krandinit(self, xp):
             else:
                 rng = np.random.RandomState(1234)
 
-            init = _krandinit(data, k, rng, xp)
-            orig_cov = xp.cov(data.T)
-            init_cov = xp.cov(init.T)
+            init = _krandinit(data, k, rng, xp_test)
+            orig_cov = cov(data.T, xp=xp_test)
+            init_cov = cov(init.T, xp=xp_test)
```

Comment:
Re: requiring input to `cov` and `_krandinit` to be from `array_api_compat` rather than raw `xp`-- this definitely had me thinking for a few minutes.

I think I concluded it makes sense to only add the shims in this one test because:
- both `_krandinit` and `cov` are rarely used in SciPy, and the former is private
- in public control flows, I'm pretty sure `_krandinit` is guaranteed to have its data coerced through the compat layer anyway (based on the source and a few local tests)

On the other hand, I don't know if there's a substantial drawback to allowing our vendored `cov` to coerce to the compat namespace -- the public control flows it operates in need that anyway I think.

Reply:
just revisiting this - I would be happy to just let `cov` coerce. I included this to be consistent with `copy` after the performance concern in https://github.com/scipy/scipy/pull/19014#discussion_r1285243643, but the overhead of one `array_namespace` call may not be significant enough to matter.

Reply:
Agreed with simplifying this. This shouldn't matter too much, so I'd rather keep the logic simpler for now. If we get at too much overhead, I can think of more structural ways to fix them than this.

Reply:
Please see https://github.com/scipy/scipy/pull/19051/commits/8c72eff3495f43fb3c9f9a761453fb1e6a241fe6 for my attempt at simplifying this. It has required letting `atleast_nd` coerce too.

Reply:
That simplification LGTM, resolving.